File: code-coaches/medium/no-numerals/NoNumerals.swift
func replaceNumbersWithWords(in sentence: String) -> String {
    let numberWords = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"]

    let words = sentence.split(separator: " ").map { (word: Substring) -> String in
        if let number = Int(word), number <= 10 {
            return numberWords[number]
        } else {
            return String(word)
        }
    }

    return words.joined(separator: " ")
}

func main() {
    let sentence = readLine()!
	print(replaceNumbersWithWords(in: sentence))
}

main()

File: code-coaches/medium/no-numerals/NoNumerals.cs
using System;
using System.Linq;

class NoNumerals
{
    static string ReplaceNumbersWithWords(string sentence)
    {
        string[] numberWords = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"};

        string[] words = sentence.Split();
        string[] updatedWords = words.Select(word =>
            int.TryParse(word, out int number) && number <= 10 ? numberWords[number] : word
        ).ToArray();

        return string.Join(" ", updatedWords);
    }

    static void Main(string[] args)
    {
        string sentence = Console.ReadLine();
        Console.WriteLine(ReplaceNumbersWithWords(sentence));
    }
}
